### PR TITLE
i18n: en-US correction of num_keep parameter description

### PR DESCRIPTION
**Pull request by rgaricano**
i18n: en-US correction of num_keep parameter description

# Pull Request Checklist

### Note to first-time contributors: Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) and describe your changes before submitting a pull request.

**Before submitting, make sure you've checked the following:**

- [ ] **Target branch:** Please verify that the pull request targets the `dev` branch.
- [ ] **Description:** Provide a concise description of the changes made in this pull request.
- [ ] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [ ] **Documentation:** Have you updated relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs), or other documentation sources?
- [ ] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?
- [ ] **Testing:** Have you written and run sufficient tests to validate the changes?
- [ ] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [ ] **Prefix:** To clearly categorize this pull request, prefix the pull request title using one of the following:
  - **BREAKING CHANGE**: Significant changes that may affect compatibility
  - **build**: Changes that affect the build system or external dependencies
  - **ci**: Changes to our continuous integration processes or workflows
  - **chore**: Refactor, cleanup, or other non-functional code changes
  - **docs**: Documentation update or addition
  - **feat**: Introduces a new feature or enhancement to the codebase
  - **fix**: Bug fix or error correction
  - **i18n**: Internationalization or localization changes
  - **perf**: Performance improvement
  - **refactor**: Code restructuring for better maintainability, readability, or scalability
  - **style**: Changes that do not affect the meaning of the code (white space, formatting, missing semi-colons, etc.)
  - **test**: Adding missing tests or correcting existing tests
  - **WIP**: Work in progress, a temporary label for incomplete or ongoing work

# Changelog Entry

### Description

Correction of num_keep parameter,
In actual description example is indicated that "last" x tokens will be retained, that is wrong, it have to say "first" x tokens.

num_keep definition:
Tokens to Keep on Context Refresh (num_keep): Retains part of the previous conversation. It's used when the n_ctx limit is reached. A new prompt will be constructed with the first n_keep characters of the original prompt plus the second half of the output to free up space for more conversation. Example: Keeping 50 first tokens helps the model remember the main topic when refreshing.

https://github.com/open-webui/open-webui/discussions/3794#discussioncomment-12691428

### Changed

correction num_keep parameter description in en-US

### Fixed
correction num_keep parameter description in en-US

### Breaking Changes

---

### Additional Information

  https://github.com/open-webui/open-webui/discussions/3794#discussioncomment-12691428

### Contributor License Agreement

By submitting this pull request, I confirm that I have read and fully agree to the [Contributor License Agreement (CLA)](/CONTRIBUTOR_LICENSE_AGREEMENT), and I am providing my contributions under its terms.

### CODE REVIEW

**Comment by tjbck** [comment]
Any official documentations that clarify `num_keep`'s behaviour?

**Comment by rgaricano** [comment]
chatgpt?

🧠 What is n_keep?
The n_keep parameter controls how many of the initial tokens from the prompt are "kept" or preserved in the context window during autoregressive generation. These tokens are treated as fixed and will not be altered or pruned from the context window, even as new tokens are generated.

llama.cpp code?

https://github.com/ggml-org/llama.cpp/blob/9eaa51e7f08593f123f00136591179a8f5956ecd/tools/main/main.cpp#L581-L586

```
...
            if (ga_n == 1) {
                // infinite text generation via context shifting
                // if we run out of context:
                // - take the n_keep first tokens from the original prompt (via n_past)
                // - take half of the last (n_ctx - n_keep) tokens and recompute the logits in batches

...
```

--------
(on ggml-org readme & param description they doesn't specify if first or last, just it say "from initial prompt", but on code is clear!,
perhaps this lack of explicit explanation and quick reading was the origin of the misunderstanding,

https://github.com/ggml-org/llama.cpp/blob/4c9fdfbe1580a66fd7d77c77418ce2c606a29fdd/tools/server/README.md?plain=1#L423-L425
```
`n_keep`: Specify the number of tokens from the prompt to retain when the context size is exceeded and tokens need to be discarded. The number excludes the BOS token.
By default, this value is set to `0`, meaning no tokens are kept. Use `-1` to retain all tokens from the prompt.
```
https://github.com/ggml-org/llama.cpp/blob/4c9fdfbe1580a66fd7d77c77418ce2c606a29fdd/common/common.h#L230
```
    int32_t n_keep                =     0; // number of tokens to keep from initial prompt
```
-------
Ollama's num_keep func.: https://github.com/ollama/ollama/blob/c001b98087e45b7b60509127d4d2e9d9ba809444/runner/ollamarunner/runner.go#L124-L131
```
...
	if int32(len(inputs)) > s.cache.numCtx {
		discard := int32(len(inputs)) - s.cache.numCtx
		newInputs := inputs[:params.numKeep]
		newInputs = append(newInputs, inputs[params.numKeep+discard:]...)

		slog.Warn("truncating input prompt", "limit", s.cache.numCtx, "prompt", len(inputs), "keep", params.numKeep, "new", len(newInputs))
		inputs = newInputs
	}
...
```

**Comment by tjbck** [comment]
LGTM, Thanks!
